plugins/modem-manager/fu-mm-device.c

```diff
@@ -216,13 +251,20 @@ fu_mm_device_probe_default (FuDevice *device, GError **error)
 		return FALSE;
 	}
 
-	/* if we have the at port reported, get sysfs path and interface number */
 	if (self->port_at != NULL) {
 		fu_mm_utils_get_port_info (self->port_at, &device_sysfs_path, &self->port_at_ifnum, NULL);
-	} else if (self->port_qmi != NULL) {
-		fu_mm_utils_get_port_info (self->port_qmi, &device_sysfs_path, NULL, NULL);
-	} else {
-		g_warn_if_reached ();
+	}
+	if (self->port_qmi != NULL) {
+		g_autofree gchar *qmi_device_sysfs_path = NULL;
+		fu_mm_utils_get_port_info (self->port_qmi, &qmi_device_sysfs_path, &self->port_qmi_ifnum, NULL);
+		if (!device_sysfs_path && qmi_device_sysfs_path) {
```

Comment:
we tend to use != NULL and == NULL for pointers

Reply:
I try to keep that in my mind all the time, looks like I missed this one :D Changed now

Reply:
Done